nifi-commons/nifi-record/src/test/java/org/apache/nifi/serialization/record/ResultSetRecordSetTest.java

```diff
@@ -272,6 +364,175 @@ public void testCreateSchemaArrayThrowsNotSupportedException() throws SQLExcepti
+
+            if (column.getRecordFieldType() instanceof DecimalDataType) {
+                DecimalDataType ddt = (DecimalDataType) column.getRecordFieldType();
+                when(resultSetMetaData.getPrecision(column.getIndex())).thenReturn(ddt.getPrecision());
+                when(resultSetMetaData.getScale(column.getIndex())).thenReturn(ddt.getScale());
+            }
+            if (column.getSqlType() == Types.JAVA_OBJECT) {
+                indexOfBigDecimal = index + 1;
+            }
+            ++index;
+        }
+
+        // Big decimal values are necessary in order to determine precision and scale
+        when(resultSet.getBigDecimal(indexOfBigDecimal)).thenReturn(new BigDecimal(String.join("", Collections.nCopies(500, "1")) + ".1"));
+
+        // This will be handled by a dedicated branch for Java Objects, needs some further details
+        when(resultSetMetaData.getColumnClassName(indexOfBigDecimal)).thenReturn(BigDecimal.class.getName());
+    }
+
+    private List<RecordField> givenFieldsThatRequireLogicalTypes() {
+        final List<RecordField> fields = new ArrayList<>();
+        fields.add(new RecordField("decimal", RecordFieldType.DECIMAL.getDecimalDataType(30, 10)));
+        fields.add(new RecordField("date", RecordFieldType.DATE.getDataType()));
+        fields.add(new RecordField("time", RecordFieldType.TIME.getDataType()));
+        fields.add(new RecordField("timestamp", RecordFieldType.TIMESTAMP.getDataType()));
+        return fields;
+    }
+
+    private RecordSchema givenRecordSchema(TestColumn[] columns) {
+        final List<RecordField> fields = new ArrayList<>(columns.length);
+
+        for (TestColumn column : columns) {
+            fields.add(new RecordField(column.getColumnName(), column.getRecordFieldType()));
+        }
+
+        return new SimpleRecordSchema(fields);
+    }
+
+    private RecordSchema givenRecordSchemaWithOnlyStringType(TestColumn[] columns) {
+        final List<RecordField> fields = new ArrayList<>(columns.length);
+
+        for (TestColumn column : columns) {
+            fields.add(new RecordField(column.getColumnName(), RecordFieldType.STRING.getDataType()));
+        }
+
+        return new SimpleRecordSchema(fields);
+    }
+
+    private List<ArrayTestData> givenArrayTypesThatRequireLogicalTypes() {
+        List<ArrayTestData> testData = new ArrayList<>();
+        testData.add(new ArrayTestData("arrayBigDecimal",
+                new BigDecimalDummy[]{new BigDecimalDummy(), new BigDecimalDummy()}));
+        testData.add(new ArrayTestData("arrayDate",
+                new Date[]{new Date(1631809132516L), new Date(1631809132516L)}));
+        testData.add(new ArrayTestData("arrayTime",
+                new Time[]{new Time(1631809132516L), new Time(1631809132516L)}));
+        testData.add(new ArrayTestData("arrayTimestamp",
+                new Timestamp[]{new Timestamp(1631809132516L), new Timestamp(1631809132516L)}));
```

Comment:
It might be easier to use a static long for the millisecond value instead of repeating the same number multiple times.